src/libmongoc/tests/TestSuite.h

```diff
@@ -710,6 +716,14 @@ _TestSuite_AddFull (TestSuite *suite,
                     ...);
 #define TestSuite_AddFull(_suite, _name, _func, _dtor, _ctx, ...) \
    _TestSuite_AddFull (_suite, _name, _func, _dtor, _ctx, __VA_ARGS__, NULL)
+#define TestSuite_AddFullWithTestFn(                            \
```

Comment:
Nit: Would `TestSuite_AddFullWithCtxFn` or `TestSuite_AddFullWithArgFn` be a more apt name?

I interpret `_func` as the test function. And `_test_fn` is the context argument passed to `_func`.

Reply:
I do not think so.

`WithArgFn` suggests an "argument function" is being provided, which I do not think is a meaningful/useful description ("Argument to what?" "A function that provides an argument?").

`WithCtxFn` suggests the argument being provided is a "context function", but the argument provided _is_ the context ("context" referring to the argument given to a test), not a function to obtain one.

`WithTestFn` was selected because `test_fn` is [the name of the parameter](https://github.com/mongodb/mongo-c-driver/blob/96ed1d70210ec5096aaeab0adbb5ae4323291f20/src/libmongoc/tests/test-mongoc-client-session.c#L1452) used by relevant tests that expect a function pointer as an argument, paralleling how parameters are [usually named `ctx`](https://github.com/mongodb/mongo-c-driver/blob/96ed1d70210ec5096aaeab0adbb5ae4323291f20/src/libmongoc/tests/test-mongoc-client-session.c#L76) by other tests. For full symmetry, I would also rename `TestSuite_AddFull` to `TestSuite_AddFullWithCtx` rather than being implicit (as an example, not as a suggestion).